src/Mjolnir.ts

```diff
@@ -950,36 +950,33 @@ export class Mjolnir {
         await this.printBanlistChanges(changes, policyList, true);
     }
 
-    private async handleConsequence(protection: Protection, roomId: string, eventId: string, sender: string, consequence: Consequence) {
-        switch (consequence.type) {
-            case ConsequenceType.alert:
-                break;
-            case ConsequenceType.redact:
-                await this.client.redactEvent(roomId, eventId, "abuse detected");
-                break;
-            case ConsequenceType.ban:
+    private async handleConsequences(protection: Protection, roomId: string, eventId: string, sender: string, consequences: Consequence[]) {
+        for (const consequence of consequences) {
+            if (consequence.name === "alert") {
+                /* take no additional action, just print the below message to management room */
+            } else if (consequence.name === "ban") {
                 await this.client.banUser(sender, roomId, "abuse detected");
-                break;
-        }
+            } else if (consequence.name === "redact") {
+                await this.client.redactEvent(roomId, eventId, "abuse detected");
+            } else {
+                throw new Error(`unknown consequence ${consequence.name}`);
+            }
 
-        let message = `protection ${protection.name} enacting ${ConsequenceType[consequence.type]}`
-            + ` against ${htmlEscape(sender)}`
-            + ` in ${htmlEscape(roomId)}`;
-        if (consequence.reason !== undefined) {
-            // even though internally-sourced, there's no promise that `consequence.reason`
-            // will never have user-supplied information, so escape it
-            message += ` (reason: ${htmlEscape(consequence.reason)})`;
+            let message = `protection ${protection.name} enacting`
+                + ` ${consequence.name}`
+                + ` against ${htmlEscape(sender)}`
+                + ` in ${htmlEscape(roomId)}`
+                + ` (reason: ${htmlEscape(consequence.reason)})`;
+            await this.client.sendMessage(this.managementRoomId, {
+                msgtype: "m.notice",
+                body: message,
+                [CONSEQUENCE_EVENT_DATA]: {
+                    who: sender,
+                    room: roomId,
+                    types: [consequence.name],
+                }
+            });
```

Comment:
As we're in a loop here, is it going to matter if a consequence stops the others from being processed because it failed e.g. redact event or ban?